src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Performance/AvoidPotentiallyExpensiveCallWhenLoggingTests.cs

```diff
@@ -0,0 +1,3532 @@
+
+                class C
+                {
+                    void M(ILogger logger, EventId eventId, Exception exception, Func<object, Exception, string> formatter)
+                    {
+                        logger.Log(LogLevel.Debug, eventId, [|4 + 2|], exception, formatter);
+                    }
+                }
+                """;
+
+            await VerifyCSharpCodeFixAsync(source, source);
+        }
+
+        [Fact]
+        public async Task CoalesceOperation_ReportsDiagnostic_CS()
+        {
+            string source = """
+                #nullable enable
+
+                using System;
+                using Microsoft.Extensions.Logging;
+
+                class C
+                {
+                    void M(ILogger logger, EventId eventId, Exception? exception, Func<object, Exception?, string> formatter)
+                    {
+                        logger.Log(LogLevel.Debug, eventId, [|exception ?? new Exception()|], exception, formatter);
+                    }
+                }
+                """;
+
+            await VerifyCSharpCodeFixAsync(source, source);
+        }
+
+        [Fact]
+        public async Task CollectionExpressionOperation_ReportsDiagnostic_CS()
+        {
+            string source = """
+                using System;
+                using Microsoft.Extensions.Logging;
+
+                class C
+                {
+                    void M(ILogger logger, EventId eventId, Exception exception, Func<int[], Exception, string> formatter)
+                    {
+                        logger.Log(LogLevel.Debug, eventId, [|[4, 2]|], exception, formatter);
+                    }
+                }
+                """;
+
+            await VerifyCSharpCodeFixAsync(source, source, CodeAnalysis.CSharp.LanguageVersion.CSharp12);
+        }
+
+        [Fact]
+        public async Task DefaultValueOperation_ReportsDiagnostic_CS()
+        {
+            string source = """
+                using System;
+                using Microsoft.Extensions.Logging;
+
+                class C
+                {
+                    void M(ILogger logger, EventId eventId, Exception exception, Func<int, Exception, string> formatter)
+                    {
+                        logger.Log(LogLevel.Debug, eventId, [|default|], exception, formatter);
+                    }
+                }
+                """;
+
+            await VerifyCSharpCodeFixAsync(source, source);
+        }
+
+        [Fact]
+        public async Task IncrementOrDecrementOperation_ReportsDiagnostic_CS()
+        {
+            string source = """
+                using System;
+                using Microsoft.Extensions.Logging;
+
+                class C
+                {
+                    void M(ILogger logger, EventId eventId, Exception exception, Func<int, Exception, string> formatter, int input)
+                    {
+                        logger.Log(LogLevel.Debug, eventId, [|input++|], exception, formatter);
```

Comment:
In this case, wrapping the log call with a check would change behaviour. Maybe the increment is needed.

Reply:
We could exclude postfix increment and decrement.